### PR TITLE
Pass SQLAlchemy engine options to FAB based UI

### LINKED ISSUE #11394: SQLAlchemy engine configuration is not passed to FAB based UI
SQLAlchemy engine configuration is not passed to FAB based UI. Faced this issue when running Airflow with MySQL metadata store with wait_timeout = 120. Webserver is failing with Internal Server Error due to "[MySQL Server has gone away](https://docs.sqlalchemy.org/en/13/faq/connections.html#mysql-server-has-gone-away)" error. Settings like `sql_alchemy_pool_recycle` and  `sql_alchemy_pool_pre_ping` have no impact on FAB internal communication with MySQL.

**Apache Airflow version**: 1.10.12

### PR DESCRIPTION
**Pull request by michalmisiewicz**
This PR fixes #11394

### CODE REVIEW

**Comment by michalmisiewicz** [comment]
@kaxil @turbaszek @mik-laj

File: airflow/settings.py [hunk 1]
```diff
@@ -158,8 +158,30 @@ def configure_orm(disable_connection_pool=False):
...
+        connect_args = {}
+
+    engine = create_engine(SQL_ALCHEMY_CONN, connect_args=connect_args, **engine_args)
+    setup_event_handlers(engine)
+
+    Session = scoped_session(
+        sessionmaker(autocommit=False,
+                     autoflush=False,
+                     bind=engine,
+                     expire_on_commit=False))
```
Comment (turbaszek) [inline airflow/settings.py]:
```suggestion
    Session = scoped_session(sessionmaker(
        autocommit=False,
        autoflush=False,
        bind=engine,
        expire_on_commit=False,
    ))
```
WDYT?

**Review by turbaszek — APPROVED** [review]

File: airflow/settings.py (hunk 1 above)
Comment (michalmisiewicz) [reply airflow/settings.py]: 👍

**Review by kaxil — COMMENTED** [review]
LGTM, needs atleast a test to check when config is set, if is reflected in flask_app.config

**Comment by michalmisiewicz** [comment]
@kaxil Test added

**Comment by michalmisiewicz** [comment]
@kaxil @turbaszek Are we ready for merge ?

**Review by kaxil — COMMENTED** [review]
Good work

**Pull request merged**